CMakeLists.txt

```diff
@@ -70,6 +70,27 @@ include_directories(${CMAKE_BINARY_DIR}/include) # just to enable vscode intelli
 #set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -march=native -mtune=native -flto") 
 add_definitions(-Wno-ignored-attributes) # to skip postgresql warnings.
 
+# Generate content type definition header file
+add_custom_target(
+    content_types ALL
+    DEPENDS "${CMAKE_BINARY_DIR}/include/li/http_server/content_types.hh"
+)
+
+add_custom_command(
+    OUTPUT "${CMAKE_BINARY_DIR}/include/li/http_server/content_types.hh"
```

Comment:
Could you use a multiline string to encode the file ? thils would help readability.
it seems like it's possible with cmake: https://cmake.org/cmake/help/v3.0/manual/cmake-language.7.html#quoted-argument

Reply:
Will do!

Reply:
Please see my latest change. Hope I understood correctly.

Reply:
Sorry I was not clear. I meant merging all the 
```
echo \"// This file is generated do not edit it.\" >> ${GENERATED_CONTENT_TYPES}
    COMMAND echo \"\#pragma once\" >> ${GENERATED_CONTENT_TYPES}
    COMMAND echo \"\#include <unordered_map>\" >> ${GENERATED_CONTENT_TYPES}
    COMMAND echo \"\#include <string_view>\" >> ${GENERATED_CONTENT_TYPES}
    COMMAND echo \"namespace li {\" >> ${GENERATED_CONTENT_TYPES}
    COMMAND echo \"static std::unordered_map<std::string_view, std::string_view> content_types = {\" >> ${GENERATED_CONTENT_TYPES}
```

in one echo command, but I realize that it does not really matter as your code is already readable enough. I'll merge it as it is.